lib/Attribute.js

```diff
@@ -56,10 +56,11 @@ function Attribute(schema, name, value) {
   else if (this.type.name === 'list'){
 
     if(value.type) {
+      if (!value.list) throw new errors.SchemaError('No list schema provided for attribute:' + this.name );
       value = value.list;
     }
 
-    if (value === undefined && value[0] === undefined){
+    if (value === undefined || value[0] === undefined){
```

Comment:
Since you said this was failing before, can we write a test that will fail in the current version, and pass with this PR to ensure this will work correctly in the future?

Reply:
Will do.

Reply:
I'm now running into an issue where if I fetch a record from dynamo that has a field set to an empty array, this error gets raised. Is that the intended behavior?

Reply:
@dobrynin Honestly I'm not sure. Are you running into that on the NPM version or only after this change? I believe this code should only be run when creating or setting up the schema/model. So I'm not sure why fetching a record from Dynamo would cause this code to be run unless you have `saveUnknown` set to `true`.

Can you try logging some things out such as `value` and see what that prints?

You can also run `export DEBUG=*` which will give you more console logs and details about what Dynamoose is doing behind the scenes.

Reply:
You’re thinking was right, we do set `saveUnknown` to `true`.